drivers: hwinfo: add wakeup flag cause

Desvauxm-st opened the pull request:
add retrieve wakeup flag of device and Clear wakeup flag

Desvauxm-st committed:
samples: boards: stm32: pm: standby_shutdown: use hwinfo_get_reset_cause()

use hwinfo_get_reset_cause() instead of LL_PWR_IsActiveFlag_SB()
remove LL_PWR_ClearFlag_WU() not  mandatory

Signed-off-by: Marc Desvaux <marc.desvaux-ext@st.com>

erwango reviewed (changes requested):
Please squash the 3 first commits.
Otherwsie LGTM

Desvauxm-st commented on drivers/hwinfo/hwinfo_stm32.c:
```diff
@@ -110,7 +116,7 @@ int z_impl_hwinfo_get_reset_cause(uint32_t *cause)
 {
 	LL_RCC_ClearResetFlags();
-
+	LL_PWR_ClearFlag_SB();
```
OK

alexanderwachter commented:
@Desvauxm-st merge commits are not allowed in Zephyr. You need to rebase your commits on main instead.

Desvauxm-st committed:
drivers: hwinfo: modification into hwinfo_get_reset_cause()

add retrieve flag SB into hwinfo_get_reset_cause()
add Clear flag SB into clear_reset_cause()

Signed-off-by: Marc Desvaux <marc.desvaux-ext@st.com>

KloudJack commented:
Since this PR was merged, unit tests for STM32 hwinfo now fail.

See the [workflow result](https://github.com/zephyrproject-rtos/zephyr/pull/56330/checks?check_run_id=12397402575) for a PR I've rebased recently.
